routers/repo/repo.go

```diff
@@ -81,6 +81,16 @@ func checkContextUser(ctx *context.Context, uid int64) *models.User {
 	return org
 }
 
+func getRepoPrivate(ctx *context.Context) bool {
+	switch setting.Repository.DefaultPrivate {
+	case setting.RepoCreatingLastUserVisibility:
+		return ctx.User.LastRepoVisibility
+	case setting.RepoCreatingPrivate:
+		return true
+	}
+	return false
```

Comment:
nit: add explicit case for `RepoCreatingPublic`, and add a `default` case in case `setting.Repository.DefaultPrivate` is something unexpected (maybe we will add more cases in the future)?